numpy/ma/core.py

```diff
@@ -3105,15 +3105,18 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
-        # 2016.01.15 -- v1.11.0
-        warnings.warn(
-            "Currently, slicing will try to return a view of the data," +
-            " but will return a copy of the mask. In the future, it will try" +
-            " to return both as views.",
-            FutureWarning
-        )
-
         dout = self.data[indx]
+
+        if (isinstance(dout, np.ndarray) and
+                np.may_share_memory(dout, self.data)):
```

Comment:
No, don't think so, unless the isinstance check can be wrong, the bounds of an indexed array parent always include the child.